### ISSUE TITLE
Have similar default values for min_delta in ReduceLROnPlateau and EarlyStopping callbacks

### ISSUE THREAD

**Issue by FlorianMerkle** [tensorflow/tensorflow#43100]
The min_delta default values are 1e-4 for the ReduceLROnPlateau callback and 0 for the EarlyStopping callback. 

This leads to cases in which training is stopped before the learning rate is reduced even though the patience parameter is higher for EarlyStopping, when the min_delta value is not set explicitly.

**Comment by ravikyram** [comment]
@FlorianMerkle 

Request you to fill [issue template.](https://github.com/tensorflow/tensorflow/issues/new/choose).

Please, share colab link or simple standalone code to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by FlorianMerkle** [comment]
@ravikyram 
I was not sure which template is appropriate, as this is not a bug but I think the default values should be consistent.

Also, this is hard to reproduce but consider the toy model with the training procedure showcased in this gist:
https://gist.github.com/FlorianMerkle/97895acf0159c45e2950b44c95d31b63#file-min_callbacks_example-py-L1

If the validation loss decreases in the patience period, but the decrease is less then 1e-4, training is stopped w/o decaying the LR first. 

I know this case is quite unlikely but I just encountered it and it took me some time to figure this out.

ReduceLROnPlateau:
https://github.com/tensorflow/tensorflow/blob/b36436b087bd8e8701ef51718179037cccdfc26e/tensorflow/python/keras/callbacks.py#L2312
EarlyStopping:
https://github.com/tensorflow/tensorflow/blob/b36436b087bd8e8701ef51718179037cccdfc26e/tensorflow/python/keras/callbacks.py#L1612

**Comment by ravikyram** [comment]
@FlorianMerkle 

I have tried in colab with TF version 2.3, nightly version(`2.4.0-dev20200910`) .Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/88de454f148bba811267d0d27be2dec2/untitled332.ipynb).You are also seeing the same behavior?
Thanks!

**Comment by jvishnuvardhan** [comment]
@FlorianMerkle Is this still an issue for you? I ran your code and don't see any issue. If default `min_delta` is not working for your model, then you can pass them as arguments 

```
early_stopping = tf.keras.callbacks.EarlyStopping(
      patience=4,
      monitor='val_loss', ,min_delta=0.04 
      )
```

Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/61c833bd1a4a2e02ec29c3339cb53e25/untitled.ipynb). 

Can you please explain little more about the concern and/or your use-case. Thanks!

**Comment by FlorianMerkle** [comment]
@jvishnuvardhan not really an issue, I just thought it is counterintuitive to have different default values for min_delta in the two callbacks. When NOT specifying values for min_delta, it can rarely occur that training is stopped early, without reducing the Learning Rate first, which is an undesired behavior imo. See my explanation from my last post:

> If the validation loss decreases in the patience period, but the decrease is less then 1e-4, training is stopped w/o decaying the LR first.
> 
> I know this case is quite unlikely but I just encountered it and it took me some time to figure this out.


I believe the default values should be 0 for both callbacks.

**Comment by jvishnuvardhan** [comment]
@FlorianMerkle I think they set these numbers as default after analyzing lot of use cases. I am not sure how much back testing we need to do make those changes. One thing we can do is, add a note on the callback page about your observations so that others are informed about selection of `min_delta`. 

Please feel free to raise a PR to update the docs. Thanks!
